web3/utils/contracts.py

```diff
@@ -164,24 +172,32 @@ def prepare_transaction(abi,
         fn_name,
         fn_args,
         fn_kwargs,
+        fn_is_fallback
     )
     return prepared_transaction
 
 
-def encode_transaction_data(abi, web3, fn_name, args=None, kwargs=None):
+def encode_transaction_data(abi, web3, fn_name, args=None, kwargs=None, is_fallback_fn=False):
     fn_abi, fn_selector, fn_arguments = get_function_info(
-        abi, fn_name, args, kwargs,
+        abi, fn_name, args, kwargs, is_fallback_fn
     )
     return add_0x_prefix(encode_abi(web3, fn_abi, fn_arguments, fn_selector))
 
 
-def get_function_info(abi, fn_name, args=None, kwargs=None):
+def get_function_info(abi, fn_name, args=None, kwargs=None, is_fallback_fn=False):
```

Comment:
I think that rather than making `get_function_info` support retrieving the fallback function you should just do it one level higher in `encode_transaction_data`.